From a476636f5bca02180f119612668eae56b18e60d8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Wed, 9 May 2012 12:39:28 +0100
Subject: [PATCH 04/14] Resolves: rhbz#805743 a11y: call doShow after we have
 a valid view

Basically a regression from ec240eafe9b25620383aa54015f5c55e0f64227a but things
were broken before that too.

enable a11y (for gnome 3 this is gsettings set org.gnome.desktop.interface
toolkit-accessibility true), and run the smoketest with dbg-util before
touching this stuff

Change-Id: I I55f4e22d8ac61c7ac3ccc37b82794604b28a8f14
---
 sd/source/ui/slidesorter/shell/SlideSorterViewShell.cxx |    2 +-
 sd/source/ui/toolpanel/ToolPanelViewShell.cxx           |    2 +-
 sd/source/ui/view/drviewsa.cxx                          |    3 +--
 sd/source/ui/view/outlnvsh.cxx                          |    4 ++--
 4 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/sd/source/ui/slidesorter/shell/SlideSorterViewShell.cxx b/sd/source/ui/slidesorter/shell/SlideSorterViewShell.cxx
index 4ef5cf6..e4075ee 100644
--- a/sd/source/ui/slidesorter/shell/SlideSorterViewShell.cxx
+++ b/sd/source/ui/slidesorter/shell/SlideSorterViewShell.cxx
@@ -187,7 +187,7 @@ void SlideSorterViewShell::Initialize (void)
         mpScrollBarBox);
     mpView = &mpSlideSorter->GetView();
 
-    ViewShell::doShow();
+    doShow();
 
     SetPool( &GetDoc()->GetPool() );
     SetUndoManager( GetDoc()->GetDocSh()->GetUndoManager() );
diff --git a/sd/source/ui/toolpanel/ToolPanelViewShell.cxx b/sd/source/ui/toolpanel/ToolPanelViewShell.cxx
index 7f01523..c890ccf 100644
--- a/sd/source/ui/toolpanel/ToolPanelViewShell.cxx
+++ b/sd/source/ui/toolpanel/ToolPanelViewShell.cxx
@@ -483,7 +483,7 @@ ToolPanelViewShell::ToolPanelViewShell( SfxViewFrame* pFrame, ViewShellBase& rVi
     ,mpSubShellManager()
     ,mnMenuId(0)
 {
-    ViewShell::doShow();
+    doShow();
 
     meShellType = ST_TASK_PANE;
 
diff --git a/sd/source/ui/view/drviewsa.cxx b/sd/source/ui/view/drviewsa.cxx
index 96c2259..54aa402 100644
--- a/sd/source/ui/view/drviewsa.cxx
+++ b/sd/source/ui/view/drviewsa.cxx
@@ -141,13 +141,12 @@ DrawViewShell::DrawViewShell( SfxViewFrame* pFrame, ViewShellBase& rViewShellBas
 , mbIsLayerModeActive(false)
 , mbIsInSwitchPage(false)
 {
-    ViewShell::doShow();
-
     if (pFrameViewArgument != NULL)
         mpFrameView = pFrameViewArgument;
     else
         mpFrameView = new FrameView(GetDoc());
     Construct(GetDocSh(), ePageKind);
+    doShow();
 }
 
 /*************************************************************************
diff --git a/sd/source/ui/view/outlnvsh.cxx b/sd/source/ui/view/outlnvsh.cxx
index 8781d60..1f4613c 100644
--- a/sd/source/ui/view/outlnvsh.cxx
+++ b/sd/source/ui/view/outlnvsh.cxx
@@ -223,8 +223,6 @@ OutlineViewShell::OutlineViewShell (
       mbInitialized(false)
 
 {
-    ViewShell::doShow();
-
     if (pFrameViewArgument != NULL)
         mpFrameView = pFrameViewArgument;
     else
@@ -233,6 +231,8 @@ OutlineViewShell::OutlineViewShell (
     mpFrameView->Connect();
 
     Construct(GetDocSh());
+
+    doShow();
 }
 
 /*************************************************************************
-- 
1.7.10.4

